ecsclient/common/token_request.py

```diff
@@ -101,10 +101,14 @@ def get_token(self):
             log.debug("Validating token")
             req = self._request(token, self.token_verification_endpoint)
 
-            if req.status_code == requests.codes.ok:
-                log.debug("Token is valid")
-                return token
+            if req.status_code == 401 or req.status_code == 403:
```

Comment:
BTW, @open4storage have you checked what status code ECS returns when the token is invalid/expired. It should only be `401 Unauthorized`, not 403. `403 Forbidden` means that the token is valid (user is authenticated) but the user has no permissions to access that resource.